src/optimizer/plugins/SimplifyArrayIndexingOptimizer.ts

```diff
@@ -0,0 +1,123 @@
+import { fshrules, fhirtypes, utils } from 'fsh-sushi';
+import { OptimizerPlugin } from '../OptimizerPlugin';
+import { Package } from '../../processor';
+import { ExportableCaretValueRule } from '../../exportable';
+
+const singletonArrayElements: Map<string, fhirtypes.PathPart[]> = new Map();
```

Comment:
I'm not sure I entirely understand why this needs to be a global variable. Doesn't that mean it will carry information between optimizations of different resources? Is that something we need?

Reply:
Addressed in [`d6e165e`](https://github.com/FHIR/GoFSH/pull/106/commits/d6e165e5611b739e6fae61fe8ba1ca39183768ae).